### PR TITLE
Add BBE for XA Transactions

### PR DESCRIPTION
**Pull request by lasinicl**

### CODE REVIEW

File: examples/error-handling-in-single-place/error_handling_in_single_place.out [hunk 1]
```diff
@@ -4,5 +4,3 @@ $ ballerina run error_handling_in_single_place.bal
 Error caught: InvalidAccountID, Account ID: -1
 Error caught: AccountNotFound, Account ID: 200
 Error caught during parsing: {ballerina/lang.int}NumberParsingError
-[ballerina/http] started HTTP/WS listener 192.168.1.18:34525
```
Comment (gimantha) [inline examples/error-handling-in-single-place/error_handling_in_single_place.out:7]: These lines will be printed again right?

Comment (lasinicl) [reply examples/error-handling-in-single-place/error_handling_in_single_place.out:7]: No. This is the output of the on fail BBE. It doesn't use JDBC.

File: ballerina/build.gradle [hunk 2]
```diff
@@ -647,6 +647,7 @@ task testExamples() {
             'the-main-function', //Needs user input
             'local-transactions',  // Xid varies
             'local-transactions-with-handlers', // Xid varies
+            'xa-transactions', // Consists transaction coordinator start log
```
Comment (praneesha) [inline ballerina/build.gradle:650]:
```suggestion
            'xa-transactions', // Consists of the transaction coordinator start log
```

File: examples/xa-transactions/tests/xa_transactions_test.bal [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import ballerina/test;
+
+(any|error)[] outputs = [];
+
+// This is the mock function which will replace the real function
```
Comment (praneesha) [inline examples/xa-transactions/tests/xa_transactions_test.bal:5]:
```suggestion
// This is the mock function, which will replace the real function.
```

File: examples/xa-transactions/tests/xa_transactions_test.bal [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+public function mockPrint(any|error... s) {
+    foreach var entry in s {
+        outputs.push(entry.toString());
+    }
+}
+
+@test:Config {}
+function testFunc() {
+    // Invoking the main function
```
Comment (praneesha) [inline examples/xa-transactions/tests/xa_transactions_test.bal:18]:
```suggestion
    // Invoking the main function.
```

File: examples/xa-transactions/xa_transactions.bal [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                            "./xa-transactions/testdb1", user = "test",
+                            password = "test", options =
+                            {datasourceName: xaDatasourceName});
+    // The JDBC Client for the second H2 database.
+    jdbc:Client dbClient2 = check new (url = "jdbc:h2:file:" +
+                            "./xa-transactions/testdb2", user = "test",
+                            password = "test", options =
+                            {datasourceName: xaDatasourceName});
+
+    // Create `Employee` table in the first database.
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.bal:19]:
```suggestion
    // Create the `Employee` table in the first database.
```

File: examples/xa-transactions/xa_transactions.bal [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    _ = check dbClient1->execute("CREATE TABLE IF NOT EXISTS EMPLOYEE " +
+                                  "(ID INT, NAME VARCHAR(30))");
+    // Create `Salary` table in the second database.
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.bal:22]:
```suggestion
    // Create the `Salary` table in the second database.
```

File: examples/xa-transactions/xa_transactions.bal [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    _ = check dbClient2->execute("CREATE TABLE IF NOT EXISTS SALARY " +
+                                 "(ID INT, VALUE FLOAT)");
+
+    // Populate tables with records.
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.bal:26]:
```suggestion
    // Populate the tables with the records.
```

File: examples/xa-transactions/xa_transactions.bal [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    var e1 = check dbClient1->execute("INSERT INTO EMPLOYEE(NAME) " +
+                                      "VALUES ('Anne')");
+    var e2 = check dbClient2->execute("INSERT INTO SALARY " +
+                                      "VALUES (1, 25000.00)");
+
+    // The transaction block initiates the transaction.
+    transaction {
+        // Execute database operations within the transaction
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.bal:34]:
```suggestion
        // Execute the database operations within the transaction
```

File: examples/xa-transactions/xa_transactions.bal [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        // to update records in `Employee` table and `Salary` table.
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.bal:35]:
```suggestion
        // to update records in the `Employee` and `Salary` tables.
```

File: examples/xa-transactions/xa_transactions.description [hunk 10]
```diff
@@ -0,0 +1,4 @@
+// Ballerina XA transactions are used when the transaction is happening over two or more databases.
+// This example uses two H2 databases which are created when executing the example. <br/><br/>
```
Comment (praneesha) [inline examples/xa-transactions/xa_transactions.description:2]:
```suggestion
// This example uses two H2 databases, which are created when executing the example. <br/><br/>
```

**Review by mohanvive — APPROVED** [review]

**Pull request merged**